### PR TITLE
Part B: Allow business processes without uploads.

### PR DESCRIPTION
**Pull request by mtuchowski**
This is in response to https://github.com/egovernment/eregistrations-salvador/pull/1737 that allow this situation.

### CODE REVIEW

File: routes/user.js [hunk 1]
```diff
@@ -32,9 +33,7 @@ module.exports = {
 	'business-process/[0-9][a-z0-9]+/documents': {
 		match: function (businessProcessId) {
 			if (!matchBusinessProcess.call(this, businessProcessId)) return false;
-			var firstUpload = this.businessProcess.requirementUploads.dataSnapshot.resolved[0];
-			if (!firstUpload) return false;
-			return matchUpload.call(this, 'requirementUpload', firstUpload.uniqueKey);
+			return matchFirstRequirementUpload.call(this);
```
Comment (kamsi) [inline routes/user.js]: No payment receipts nor certificates possible here?

File: view/business-process-official.js [hunk 2]
```diff
@@ -11,11 +11,17 @@ exports._match = 'businessProcess';
 exports['sub-main'] = {
 	class: { content: true, 'user-forms': true },
 	content: function () {
+		var hasDocuments = or(
+			this.businessProcess.requirementUploads.applicable._size,
+			this.businessProcess.paymentReceiptUploads.applicable._size,
+			this.businessProcess.certificates.userApplicable._size
```
Comment (kamsi) [inline view/business-process-official.js]: I'm not sure that we should include `userApplicable` collection for certificates in context of official. I think that for official it should be `applicable`.

File: view/business-process-revision.js [hunk 3]
```diff
@@ -168,64 +168,87 @@ exports._pauseButton = function (/*options*/) {
...
 exports._revisionContainer = function () {
-	var paymentUploads = getUploads(this.processingStep.paymentReceiptUploads, this.appName);
-	section({ class: 'section-tab-nav' },
-		a({
+	var requirementUploads    = this.processingStep.requirementUploads
+	  , paymentReceiptUploads = this.processingStep.paymentReceiptUploads
+	  , uploads               = getUploads(requirementUploads, this.appName)
+	  , paymentUploads        = getUploads(paymentReceiptUploads, this.appName);
+
+	var paymentUploadsTabNumber = resolve(uploads, '_length').map(function (uploadsLength) {
```
Comment (kamsi) [inline view/business-process-revision.js]: Let's keep `paymentUploadsTabNumber` as an observable which stores a number. We should have observable for each tab number. And they should actually store numbers. One should calculate it's number as previous + 1. Then, once you pass those observables to i18n placeholders use `map` to transform the number value to string.

File: routes/user.js (hunk 1 above)
Comment (mtuchowski) [inline routes/user.js]: Nope, never was.

**Pull request merged**